### PR TITLE
getWorld optimization

### PR DESCRIPTION
**Pull request by otradovec**
Two occurrences of for loop, which is executed max once >> changed to if

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

**Comment by Cervator** [comment]
Hey again @otradovec - I'm not sure this code change is an improvement. It is a bit more awkward code for the same effect, even if the `for` loop itself is kinda goofy/wasteful when the _expectation_ is for there to be just one world entity. Probably we ultimately need to improve the backing code to where we either have the option to return a singleton entity or better handle the _potential_ for multiple (as we'll support multiple worlds sooner or later)

In other words it makes already awkward code that needs to be refactored fundamentally into more _explicitly_ awkward (if technically more accurate) code, if that makes sense. Chances are we do this in many places over the codebase, and really should just have a better way in general to handle singleton entities :-)

Curious if @immortius would have a comment for how we should architecturally approach singleton entities.

**Comment by otradovec** [comment]
@Cervator
I have not though about having more worlds, then it makes more sence. Something like nether? I have never played this game, because I wasn't able to run it, but the code is interesting.

**Comment by Cervator** [comment]
Yep exactly 👍 You did find some code that _does_ need improvement, just not quite sure how to handle it yet :-)
